egui_demo_lib/src/apps/demo/plot_demo.rs

```diff
@@ -366,45 +371,62 @@ impl Widget for &mut ItemsDemo {
             plot_ui.image(image.name("Image"));
             plot_ui.arrows(arrows.name("Arrows"));
         })
+        .response
     }
 }
 
 #[derive(PartialEq)]
-struct InteractionDemo {
-    pointer_coordinate: Option<Value>,
-    pointer_coordinate_drag_delta: Vec2,
-}
+struct InteractionDemo {}
 
 impl Default for InteractionDemo {
     fn default() -> Self {
-        Self {
-            pointer_coordinate: None,
-            pointer_coordinate_drag_delta: Vec2::ZERO,
-        }
+        Self {}
     }
 }
 
 impl Widget for &mut InteractionDemo {
     fn ui(self, ui: &mut Ui) -> Response {
-        let coordinate_text = if let Some(coordinate) = self.pointer_coordinate {
-            format!("x: {:.03}, y: {:.03}", coordinate.x, coordinate.y)
+        let plot = Plot::new("interaction_demo").height(300.0);
+
+        let InnerResponse {
+            response,
+            inner: (screen_pos, pointer_coordinate, pointer_coordinate_drag_delta, bounds, hovered),
+        } = plot.show(ui, |plot_ui| {
+            (
+                plot_ui.screen_from_plot(Value::new(0.0, 0.0)),
+                plot_ui.pointer_coordinate(),
+                plot_ui.pointer_coordinate_drag_delta(),
+                plot_ui.plot_bounds(),
+                plot_ui.plot_hovered(),
+            )
+        });
+
+        ui.label(format!(
+            "plot bounds: min: {:.02?}, max: {:.02?}",
```

Comment:
I'm curious: what does the `?` accomplish here? Is the debug-formatting of a `f64` different from its default formatting?

Reply:
Those are arrays, so the debug printing is needed. I was quite surprised though to see that the float formatting works as well in that case.

Reply:
oh right!